cosmwasm/osmosis-tokenfactory-token-minter/src/contract.rs

```diff
@@ -0,0 +1,315 @@
+                                        DenomUnit {
+                                            denom: metadata.symbol.clone(),
+                                            exponent: metadata.decimals.into(),
+                                            aliases: vec![],
+                                        },
+                                    ],
+                                    base: denom.clone(),
+                                    display: metadata.symbol.clone(),
+                                    name: metadata.name,
+                                    symbol: metadata.symbol,
+                                    uri: "".to_string(),
+                                    uri_hash: "".to_string(),
+                                }),
+                            }
+                            .encode_to_vec()
+                            .into(),
+                        },
+                        CosmosMsg::Custom(TokenFactoryMsg::ChangeAdmin {
+                            denom,
+                            new_admin_address: TOKEN_ADMIN.load(deps.storage)?.to_string(),
+                        }),
+                    ]
+                }
+                WrappedTokenMsg::MintTokens {
+                    denom,
+                    amount,
+                    mint_to_address,
+                } => {
+                    vec![CosmosMsg::Custom(TokenFactoryMsg::MintTokens {
+                        denom,
+                        amount,
+                        mint_to_address,
+                    })]
+                }
+                WrappedTokenMsg::BurnTokens {
+                    denom,
+                    amount,
+                    burn_from_address,
+                    ..
+                } => {
+                    let contains_base_token = info
+                        .funds
+                        .iter()
+                        .any(|coin| coin.denom == denom && coin.amount == amount);
+                    if !contains_base_token {
+                        return Err(Error::MissingFunds {
+                            denom: denom.clone(),
+                            amount,
+                        });
+                    }
+                    vec![CosmosMsg::Custom(TokenFactoryMsg::BurnTokens {
+                        denom,
+                        amount,
+                        burn_from_address,
+                    })]
+                }
+            };
+            Response::new().add_messages(msgs)
+        }
+        ExecuteMsg::Local(msg) => match msg {
+            LocalTokenMsg::Escrow { denom, amount, .. } => {
+                let contains_base_token = info
+                    .funds
+                    .iter()
+                    .any(|coin| coin.denom == denom && coin.amount == amount);
+                if !contains_base_token {
+                    return Err(Error::MissingFunds { denom, amount });
+                }
+                Response::new()
+            }
```

Comment:
bbn12z2k2wajxty2pqhh9znwn9tusrupkqj7fsdh37

Reply:
In zkgm, we always pass the given pass to the minter and if the minter is native, it checks whether the funds are provided.